api/src/com/cloud/agent/api/to/S3TO.java

```diff
@@ -118,6 +122,14 @@ public boolean equals(final Object thatObject) {
             return false;
         }
 
+        if (connectionTtl != null ? !connectionTtl.equals(thatS3TO.connectionTtl) : thatS3TO.connectionTtl != null) {
+            return false;
```

Comment:
I agree, but the other lines there also have the very complex if-else statements.

I want to write the S3 code anyway, it's buggy as hell